core/compiler.cpp

```diff
@@ -108,17 +95,19 @@ void AddEngineToGraph(
   g->block()->appendNode(unpack_node);
 
   // If there are multiple output tensors from TensorRT we wrap them in a tuple
-  // to return
-  if (unpack_node->outputs().size() > 1) {
+  // to return, convert to tuple only when we only have 1 segmented graph
+  if (!engine_id && unpack_node->outputs().size() > 1) {
```

Comment:
Does the case where we have multiple TRT engines never have engine_id 0? We should not this if so.

Reply:
Yes. I refactored this function by adding a default argument indicating whether we have fallback or not.